tools/mapping/gen_mapping_files/runoff_to_ocn/src/smooth_mod.F90

```diff
@@ -26,13 +21,15 @@ MODULE smooth_mod
 CONTAINS
 !===============================================================================
 
-SUBROUTINE smooth_init(map_in, map_out)
+SUBROUTINE smooth_init(ofilename, restrict_smooth_src_to_nn_dest,  map_in, map_out)
 
    implicit none
 
    !--- arguments ---
-   type(sMatrix),intent( in)   :: map_in   ! original unsmoothed, matrix
-   type(sMatrix),intent(inout) :: map_out  ! smoothing matrix
+   character(*), intent(in)    :: ofilename                       ! name of ocn scrip grid file
+   type(sMatrix),intent(in)    :: map_in                          ! original unsmoothed, matrix
+   logical,      intent(in)    :: restrict_smooth_src_to_nn_dest  ! map_out%mask_a = pts mapped to map_in
```

Comment:
Can you add some more comments, either here or in the PURPOSE comment below, describing why you might choose to have restrict_smooth_src_to_nn_dest false instead of true? Something along the lines of what you have in the README, perhaps.